digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/csvDataExporter/AssetReportCsvExporter.scala

```diff
@@ -50,14 +50,27 @@ class AssetReportCsvExporter(roadLinkServiceImpl: RoadLinkService, eventBusImpl:
     }
 
     def executeQuery(municipality: Int): List[AssetReport] = {
+      val linkIds = roadLinkServiceImpl.getVVHRoadLinksF(municipality)
+        .filter(_.administrativeClass == Municipality)
+        .map(_.linkId)
+
       if (extractPointsAsset) {
-        assetReporterDAO.pointAssetQuery(Seq(municipality), assetTypesList)
-      }
-      else {
-        val linkIds = roadLinkServiceImpl.getVVHRoadLinksF(municipality)
-                                         .map(_.linkId)
+        if(assetTypesList.size == 1 && assetTypesList.head == ServicePoints.typeId) {
+          assetReporterDAO.servicePointQuery(Seq(municipality))
+        } else {
+          val pointAssetReports = assetReporterDAO.pointAssetQuery(linkIds, assetTypesList.filterNot(_ == ServicePoints.typeId))
+
+          val servicePointAssetReports = if(assetTypesList.contains(ServicePoints.typeId)) assetReporterDAO.servicePointQuery(Seq(municipality)) else List()
+
+          pointAssetReports ++ servicePointAssetReports
+        }
```

Comment:
At this point, we don't need to perform the second IF clause if we modify the query inside of the pointAssetQuery method, and the "filterNot(_ == ServicePoints.typeId)" can be deleted too.

Because, since the service points don't have any relation with LRM_POSITION, the query inside of the pointAssetQuery method will return nothing when the ServicePoints type id is selected.

Reply:
Applied